src/disco/sbpf/fd_opcode.h

```diff
@@ -0,0 +1,226 @@
+#ifndef HEADER_fd_src_disco_BPF_fd_opcode_h
+#define HEADER_fd_src_disco_BPF_fd_opcode_h
+
+/* This header defines the opcodes used in the sBPF (Solana BPF)
+   instruction set. The opcode is the first byte of each instruction
+   slot. */
+
+/* Opcode encoding helpers ********************************************/
+
+/* BPF_OPC: Opcode number given the operation class, mode, sub-operation. */
+#define BPF_OPC( class, mode, variant ) (uchar)( (BPF_CLASS_##class)|(BPF_##mode)|(BPF_##variant) )
+
+/* BPF_CLASS_{...}: Instruction class */
+#define BPF_CLASS_LD     ((uchar)0x00) /* Load                        */
+#define BPF_CLASS_LDX    ((uchar)0x01) /* Load Indexed                */
+#define BPF_CLASS_ST     ((uchar)0x02) /* Store                       */
+#define BPF_CLASS_STX    ((uchar)0x03) /* Store Indexed               */
+#define BPF_CLASS_ALU32  ((uchar)0x04) /* 32-bit integer arithmetic   */
+#define BPF_CLASS_JMP    ((uchar)0x05) /* Control flow                */
+#define BPF_CLASS_ALU64  ((uchar)0x07) /* 64-bit integer arithmetic   */
```

Comment:
In addition to the consistent naming, debating with myself whether or not these macros should have a FD_ prefix.  Default for public headers is yes to avoid collisions in user code (that could well be bringing in rando headers from all over the place).  But I suspect (like some of the stuff in simd), these macros are liable to used in pretty well encapsulated locations that the collision risk is low and then often enough in those locations that the namespace-like prefixing is a hinderance to readability.

If we decide to keep as is and not prefixing though, should add a comment at the top to explain why these aren't prefixed though.